arbor/include/arbor/morph/embed_pwlin1d.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+// Embedding of cell morphology as 1-d tree with piecewise linear radius.
+
+#include <vector>
+
+#include <arbor/morph/morphology.hpp>
+#include <arbor/morph/primitives.hpp>
+
+namespace arb {
+
+struct embed_pwlin1d_data;
+
+namespace util {
+template <typename X> struct pw_elements;
+}
+
+using pw_constant_fn = util::pw_elements<double>;
+
+struct embed_pwlin1d {
+    explicit embed_pwlin1d(const arb::morphology& m);
+
+    mlocation sample_location(msize_t sid) const {
+        return sample_locations_.at(sid);
+    }
+
+    // Interpolated radius at location.
+    double radius(mlocation) const;
+
+    // Computed length of mcable.
+    double integrate_length(mcable c) const;
+    double integrate_length(msize_t bid, const pw_constant_fn&) const;
```

Comment:
It's an alias defined a bit above for `pw_elements<double>` — is it the use of `pw_elements` to represent a piecewise constant function that needs more explanation, or `pw_elements` itself?

Could remove the alias if it helps to clarify what is going on (I added it because I thought it made things more clear).